lib/hawkular/base_client.rb

```diff
@@ -175,6 +178,16 @@ def tenant_header
       headers
     end
 
+    # @!visibility private
+    def connect_error(fault)
+      if fault.is_a?(SocketError)
+        HawkularConnectionException.new(fault.to_s)
+      elsif [Errno::ECONNREFUSED, Errno::ECONNRESET, Errno::EHOSTUNREACH,
+             Errno::ENETDOWN, Errno::ENETUNREACH, Errno::ETIMEDOUT].include?(fault.class)
+        HawkularConnectionException.new(fault.to_s, fault.class::Errno)
```

Comment:
this is very similar to line 184. Could you please elaborate what do we gain here by adding the fault.class vs the previous assignment in the "if" part where it is not added to the exception?